Week3/homework/app.js

```diff
@@ -0,0 +1,150 @@
+  database: 'todo',
+});
+
+// Create User
+const createUser = (req, res) => {
+  const user_name = req.body.name;
+  if (!user_name) {
+    return res.status(400).send('Please enter a username for user!');
+  }
+  const createUser = `INSERT INTO Users (username) VALUES (?);`;
+  connection.query(createUser, [user_name], err => {
+    if (err) {
+      return res.status(500).send('Something went wrong');
+    }
+  });
+  return res.status(201).send('New user successfully created!');
+};
+
+//Get users
+const getUsers = (req, res) => {
+  const getUsersQuery = `SELECT * from users`;
+  connection.query(getUsersQuery, (err, results) => {
+    if (err) {
+      return res.status(500).send('Something went wrong');
+    }
+    res.json(results);
+  });
+};
+
+// Get a single user
+const getUser = (req, res) => {
+  const user_id = req.params.id;
+  const getUsersQuery = `SELECT * from users WHERE user_id = ?`;
+  connection.query(getUsersQuery, [user_id], (err, results) => {
+    if (err) {
+      return res.status(500).send('Something went wrong');
+    }
+    res.json(results);
+  });
+};
+
+// Update User
+const updateUser = (req, res) => {
+  const user_id = req.params.id;
+  const user_name = req.body.name;
+  const updateUsersQuery = `UPDATE users SET username = ?  WHERE user_id = ${user_id}`;
+  connection.query(updateUsersQuery, [user_name], err => {
+    if (err) {
+      return res.status(500).send('Something went wrong');
+    }
+    res.status(200).send('user updated');
+  });
+};
+
+// Create todolist
+const createTodoList = (req, res) => {
+  const user_id = req.body.user;
+  const todolist_name = req.body.name;
+  const createTodolist = `INSERT INTO todolist (user_id, todolist_Name) VALUES (?, ?)`;
+  connection.query(createTodolist, [user_id, todolist_name], err => {
+    if (err) {
+      throw err;
+    }
+    res.status(200).send('Todolist created!');
+  });
+};
+
+// Delete todo list
+const deleteTodoList = (req, res) => {
+  const todolist_id = req.params.id;
+  const deleteTodoListQuery = `DELETE FROM todolist WHERE todolist_id = ?`;
+  connection.query(deleteTodoListQuery, [todolist_id], err => {
+    if (err) {
+      return res.status(500).send('Something went wrong');
+    }
+    res.status(200).send('Todo list deleted');
+  });
+};
+
+//Create todos list
+const createTodo = (req, res) => {
+  const todolist_id = req.body.todolist_id;
+  const todo_item = req.body.name;
+  const tag = req.body.tag;
+  const done = req.body.done;
+  const createTodo = `INSERT INTO todos (todolist_id, todo_item, tag, done ) VALUES (?, ?, ?, ?)`;
+  connection.query(createTodo, [todolist_id, todo_item, tag, done], err => {
+    if (err) {
+      return res.status(500).send('Something went wrong');
+    }
+    return res.status(201).send('Todo item created!');
+  });
+};
+
+// Detete todos item
+const deleteTodoItem = (req, res) => {
+  const todo_id = req.params.id;
+  if (!todo_id) {
+    return res.status(400).send('Todo not found!');
```

Comment:
Great error handling. Just one detail, you want 404 here I believe since the error message is "todo not found"